javascript.md

```diff
@@ -75,6 +84,8 @@ var results = do_something(3);
 
 Rust:
 
+Note again that the concept of "hoisting" doesn't really exist in Rust; all function definitions are scoped.
+
 ```rust
 // Function definition: takes an integer argument, returns an integer
 fn do_something(some_argument: i32) -> i32 {
```

Comment:
I can definitely clear up what the parts of this function definition are!

Hoooooo boy. This was [a whole thing](https://internals.rust-lang.org/t/restarting-the-int-uint-discussion/1131/31). The short story is that no, there isn't a generic "number" type-- since Rust is a systems language, you do have to think a bit about how much space you want your types to take up. I usually start with i32 since it's rare that I want numbers less than -2 ^ 31 or greater than 2 ^ 31 or that I want to optimize my program's memory usage to that amount of detail. Looks like [ES6's Number.MAX_SAFE_INTEGER is (2 ^ 53 - 1](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER)

[There's a list of all the integral types in the Reference](https://doc.rust-lang.org/stable/reference.html#integer-literals) and I can totally add a link to that somewhere in here.